### PR TITLE
Add remote config loading support to amp-analytics.

### PR DESCRIPTION
**Pull request by btownsend**
Adds support for loading remote config from a URL specified in an `amp-analytics` tag's `config` attribute.

### CODE REVIEW

File: examples/analytics.amp.html [hunk 1]
```diff
@@ -75,6 +75,8 @@
 </script>
 </amp-analytics>
 
+<amp-analytics id="analytics3" config="analytics.config.json"></amp-analytics>
```
Comment (cramforce) [inline examples/analytics.amp.html]: Make the URL `./analytics.config.json` to make it more obvious what is happening here.

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 2]
```diff
@@ -84,34 +82,65 @@ export class AmpAnalytics extends AMP.BaseElement {
...
+     */
+    this.remoteConfig = {};
+
+    return this.fetchRemoteConfig_().then(() => {
+      /**
+       * @private {!JSONObject} The analytics config associated with the tag
+       */
+      this.config_ = this.mergeConfigs_();
+
+      if (this.hasOptedOut_()) {
```
Comment (cramforce) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Just a general comment: Should this be in a more central place, so it isn't as easy to forget?

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 3]
```diff
@@ -84,34 +82,65 @@ export class AmpAnalytics extends AMP.BaseElement {
...
+   * @return {!Promise<>}
+   */
+  fetchRemoteConfig_() {
+    const remoteConfigUrl = this.element.getAttribute('config');
+    if (!remoteConfigUrl) {
+      return Promise.resolve();
     }
-    return Promise.resolve();
+    log.fine(this.getName_(), 'Fetching remote config', remoteConfigUrl);
+    return xhrFor(this.getWin()).fetchJson(remoteConfigUrl).then(jsonValue => {
```
Comment (cramforce) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Add assertHttpsUrl

**Comment by cramforce** [comment]
LGTM

File: examples/analytics.amp.html (hunk 1 above)
Comment (btownsend) [inline examples/analytics.amp.html]: Done.

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 3 above)
Comment (btownsend) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Done.

**Comment by btownsend** [comment]
Feedback addressed and commits squashed.

**Comment by btownsend** [comment]
Update to resolve conflicts with base branch.

**Comment by cramforce** [comment]
Merged. Thanks!

**Pull request merged**